itsm_api/views.py

```diff
@@ -1447,13 +1447,16 @@ def _get(self, comment_id):
             flask_restful.abort(404, message=f"Comment '{comment_id}' not found!")
         ticket_data   = DB_TICKET_TABLE.get(doc_id=comment['ticket_id'])
         customer_data = DB_CUSTOMER_TABLE.get(doc_id=ticket_data['customer_id'])
-        user_data     = DB_USER_TABLE.get(doc_id=comment['user_id'])
+        user_data = {}
+        if comment.get('user_id'):
+            user_data     = DB_USER_TABLE.get(doc_id=comment['user_id'])
         res = dict(comment)
         res.update({
             "id" : comment.doc_id,
             "_embedded" : {
                 "ticket"   : self.embed_ticket_data_in_result([ticket_data])[0],
-                "user"     : self.embed_user_data_in_result([user_data])[0],
+                "user"     : (self.embed_user_data_in_result([user_data])[0]
+                              if user_data else {}),
```

Comment:
I think instead of having this here we could just move it outside the `res.update`, then do another update if `user_data` exists:

```python
res.update({
    "id" : comment.doc_id,
    "_embedded" : {
        "ticket"   : self.embed_ticket_data_in_result([ticket_data])[0],
        "customer" : self.embed_customer_data_in_result([customer_data])[0]
    },
    '_links' : self.make_links({
                      "self"         : Comment.get_self_url(comment.doc_id),
                      "contained_in" : CommentList.get_self_url(),
                  })
})
# Only embed the user in the response if user_data exists
if user_data:
    res['_embedded'].update({"user": self.embed_user_data_in_result([user_data])[0]})

```

Since `user` will be empty anyway, the response can look a bit cleaner

Reply:
Are we expected to provide user always? I do not know what the expectation is. @jbrendel can you comment on the expectation?

Reply:
The expectation was that it's always there, since it always was associated with a user. That seems to change now, so ... I don't know what the correct behaviour is. I think it's ok to have the presence optional, if no user is known.